src/lib/mathlib/math/SearchMin.hpp

```diff
@@ -60,8 +59,8 @@ inline const _Tp goldensection(const _Tp &arg1, const _Tp &arg2, _Tp(*fun)(_Tp),
 {
 	_Tp a = arg1;
 	_Tp b = arg2;
-	_Tp c = b - (b - a) / ((_Tp)GOLDEN_RATIO);
-	_Tp d = a + (b - a) / ((_Tp)GOLDEN_RATIO);
+	_Tp c = b - (b - a) / GOLDEN_RATIO;
```

Comment:
Cast GOLDEN_RATIO to _Tp. I'm assuming to avoid the double promotion if using floats.

Reply:
The reason why I removed it is because in theory you could also use with `int`, and then I certainly don't want to cast `GOLDEN_RATIO`. For `float` it does not matter.

Reply:
but actually not sure if anyone would ever use it with `int`

Reply:
If anyone ever wanted to use it with int we could add a special case. For now the important thing is to prevent unnecessary double floating point calculations, which are an order of magnitude slower on the STM32F4 targets.